main.tf

```diff
@@ -149,7 +150,8 @@ resource "aws_autoscaling_group" "default" {
   count = module.this.enabled ? 1 : 0
 
   name_prefix               = format("%s%s", module.this.id, module.this.delimiter)
-  vpc_zone_identifier       = var.subnet_ids
+  vpc_zone_identifier       = var.availability_zones == null ? var.subnet_ids : null
+  availability_zones        = var.subnet_ids == null ? var.availability_zones : null
```

Comment:
I'm confused if this logic + additional variable is necessary. Can the user still specify a `network_interface_id` AND `subnet_ids`? I think we'd prefer to not introduce this additional logic and functionality if the same can be passed by the consumer of this module passing the correct subnet IDs. Please provide some more info here before we move this one forward. Thanks!

Reply:
From what I've seen these past few days and from what I've found on the web (https://stackoverflow.com/questions/63374810/network-interface-conflicts-with-subnet-id-terraform-aws-provider, https://ricdros.com/ASG-no-ENI-and-subnet/), the ASG doesn't work if both `network_interface_id` and `vpc_zone_identifier` are specified.
But you're right, the availability_zones variable is somewhat unnecessary in this case and we can fetch that information from a aws_subnet data source, i.e.
```hcl
data "aws_subnet" "this" {
  for_each = toset(var.subnet_ids)
  id            = each.value
}
...
resource "aws_autoscaling_group" "default" {
  vpc_zone_identifier = var.network_interface_id == null ? var.subnet_ids : null
  availability_zones    = var.network_interface_id != null ? data.aws_subnet.this[*].availability_zone : null
```

Reply:
Ah gotcha. But yeah that seems like a better solution to the problem. Mind implementing that?

Reply:
No. I was just awaiting your go-ahead. Removed the variable and used a data source and local to capture the azs.